app/controllers/modules/course/announcements_module.rb

```diff
@@ -5,8 +5,21 @@ class Course::AnnouncementsModule
     [
       {
         title: I18n.t('course.announcements.sidebar_title'),
-        unread: 0
+        unread: Course::AnnouncementsModule.get_unread_number(@current_user,@course)
       }
     ]
   end
+
+  private
+
+  def self.get_unread_number (current_user,course)
```

Comment:
This module declaration file is for our modular framework, I don't think you should declare your function here.

You might want to declare it inside announcement model.